src/libraries/Microsoft.Extensions.DependencyInjection.Abstractions/src/ActivatorUtilities.cs

```diff
@@ -311,6 +314,11 @@ private static MethodInfo GetMethodInfo<T>(Expression<T> expr)
 
             return (IServiceProvider serviceProvider, object?[]? arguments) =>
             {
+                if (serviceProvider is null)
+                {
+                    throw new ArgumentNullException(nameof(serviceProvider));
```

Comment:
Adds the same check that was added for the LINQ case: https://github.com/dotnet/runtime/pull/82739

Reply:
NIT: Don't you want to add the check for above `constructorParameters.Length == 0` case too? Also could we use `ArgumentNullException.ThrowIfNull(serviceProvider);`?

Reply:
The [ObjectFactory](https://learn.microsoft.com/dotnet/api/microsoft.extensions.dependencyinjection.objectfactory) delegate does allow `null` or zero-length array for the arguments. It just doesn't allow `null` for the `serviceProvider` because the caller should just use `Activator.CreateInstance()` or `ConstructorInfo.Invoke()` instead.

For using `ArgumentNullException.ThrowIfNull` instead, since this code also compiles for NetStandard, that isn't feasible.